lib/rules/prefer-user-event.ts

```diff
@@ -90,59 +90,38 @@ export default ESLintUtils.RuleCreator(getDocsUrl)<Options, MessageIds>({
   },
   defaultOptions: [{ allowedMethods: [] }],
 
-  create(context, [options]) {
+  create(context, [options], helpers) {
     const { allowedMethods } = options;
     const sourceCode = context.getSourceCode();
-    let hasNamedImportedFireEvent = false;
-    let hasImportedFireEvent = false;
-    let fireEventAlias: string | undefined;
-    let wildcardImportName: string | undefined;
 
     return {
-      // checks if import has shape:
-      // import { fireEvent } from '@testing-library/dom';
-      ImportDeclaration(node: TSESTree.ImportDeclaration) {
-        if (!hasTestingLibraryImportModule(node)) {
-          return;
-        }
-        const fireEventImport = node.specifiers.find(
-          (node) =>
-            isImportSpecifier(node) && node.imported.name === 'fireEvent'
-        );
-        hasNamedImportedFireEvent = !!fireEventImport;
-        if (!hasNamedImportedFireEvent) {
+      ['CallExpression > MemberExpression'](node: TSESTree.MemberExpression) {
+        if (!helpers.getIsTestingLibraryImported()) {
           return;
         }
-        fireEventAlias = fireEventImport.local.name;
-      },
+        const testingLibraryImportNode = helpers.getTestingLibraryImportNode();
+        const fireEventAliasOrWildcard = getSpecifierFromImport(
+          testingLibraryImportNode,
+          'fireEvent'
+        )?.local.name;
 
-      // checks if import has shape:
-      // import * as dom from '@testing-library/dom';
-      'ImportDeclaration ImportNamespaceSpecifier'(
-        node: TSESTree.ImportNamespaceSpecifier
-      ) {
-        const importDeclarationNode = node.parent as TSESTree.ImportDeclaration;
-        if (!hasTestingLibraryImportModule(importDeclarationNode)) {
+        if (!fireEventAliasOrWildcard) {
+          // testing library was imported, but fireEvent was not imported
           return;
         }
-        hasImportedFireEvent = !!node.local.name;
-        wildcardImportName = node.local.name;
-      },
-      ['CallExpression > MemberExpression'](node: TSESTree.MemberExpression) {
-        if (!hasImportedFireEvent && !hasNamedImportedFireEvent) {
-          return;
-        }
-        // check node is fireEvent or it's alias from the named import
         const fireEventUsed =
-          isIdentifier(node.object) && node.object.name === fireEventAlias;
+          isIdentifier(node.object) &&
+          node.object.name === fireEventAliasOrWildcard;
+
         const fireEventFromWildcardUsed =
           isMemberExpression(node.object) &&
           isIdentifier(node.object.object) &&
-          node.object.object.name === wildcardImportName &&
+          node.object.object.name === fireEventAliasOrWildcard &&
           isIdentifier(node.object.property) &&
           node.object.property.name === 'fireEvent';
```

Comment:
Shouldn't this one be `fireEventAliasOrWildcard` too?

Reply:
no, this scenario validates is for the case when rtl was imported with `import * as rtl from 'module'`.
Because of that, `fireEvent` is never aliased, as you can do `rtl.fireEvent.change()` or whatever alias you placed to the whole module.

I'm not covering

```ts
import * as rtl from 'module'
const aliasedFireEvent = rtl.fireEvent
aliasedFireEvent.change()
```

because that's too much 😅 

So `node.object.object.name === fireEventAliasOrWildcard` checks for the `rtl` or whatever alias you've used (the tests use `dom` for instance), and the line from this comment should always be `fireEvent`

Reply:
Ah of course! Thanks for the clarification.